src/main/java/io/seqera/tower/agent/Agent.java

```diff
@@ -182,17 +188,47 @@ private void sendPeriodicHeartbeat() {
     }
 
     private void sendInfoMessage() throws IOException {
-        String userName = new UnixSystem().getUsername();
-        String workDir = this.workDir.toAbsolutePath().normalize().toString();
-        String agentVersion = getVersion();
-
         agentClient.send(new InfoMessage(
-                userName,
-                workDir,
-                agentVersion
+                validatedUserName,
+                validatedWorkDir,
+                getVersion()
         ));
     }
 
+    private void validateParameters() throws IOException {
+        // Fetch username
+        validatedUserName = System.getProperty("user.name");
+        if (validatedUserName == null || validatedUserName.isEmpty() || validatedUserName.isBlank()) {
+            logger.error("Impossible to detect current Unix username.");
+            System.exit(-1);
+        }
+
+        // Set default workDir
+        if (workDir == null) {
+            String defaultPath = System.getProperty("user.home") + "/work";
```

Comment:
let's log it conditon

Reply:
What do you mean? Now I always log the `userName` and the `workDir`.
http://github.com/seqeralabs/tower-agent/blob/1c9ef357e3d26e67006b9c9921d36eee39e9d3e9/src/main/java/io/seqera/tower/agent/Agent.java#L229-L229

Reply:
was just suggesting to log the `workDir == null` condition. it can be useful for debug (?)

Reply:
Ok, added as a debug log.